### PR TITLE
Fs Upgrade suite for 5.3 to 6.0

### PR DESCRIPTION
**Pull request by AmarnatReddy**
Fs Upgrade suite for 5.3 to 6.0

Time taken to complete Suite : 54 mins, 21 secs
Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-ljejt/
Jenkins run : https://jenkins.ceph.redhat.com/view/RHCS%20QE/job/rhceph-test-executor/2467/console

Latest Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-s0pco/
Signed-off-by: Amarnath K <amk@amk.remote.csb>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: suites/quincy/cephfs/tier-0_cephfs_upgrade_53_to_6x.yaml [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                duration: 10
+              verify_cluster_health: true
+            destroy-cluster: false
+      desc: Running upgrade and i/o's parallelly
+      abort-on-fail: true
+  - test:
+      abort-on-fail: false
+      desc: Validates the data after upgrade
+      module: cephfs_upgrade.metadata_version_validation.py
+      name: "creation of Prerequisites for Upgrade"
```
Comment (yogesh-mane) [inline suites/quincy/cephfs/tier-0_cephfs_upgrade_53_to_6x.yaml:179]: Could we change the name to something like "Verification of metadata after upgrade"

**Comment by yogesh-mane** [comment]
Also metadata_version_validation.py:114 file's line 114 & 115 out is showing "None". Could you check that?

File: suites/quincy/cephfs/tier-0_cephfs_upgrade_53_to_6x.yaml [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+      abort-on-fail: false
+      desc: Creates all the fs related components before upgrade
+      module: cephfs_upgrade.upgrade_pre_req.py
+      name: "creation of Prerequisites for Upgrade"
+      polarion-id: CEPH-83575312
+  - test:
+      abort-on-fail: false
+      desc: Collects all the info required for validating the upgrade
+      module: cephfs_upgrade.metadata_version_validation.py
+      name: "Metadata validation for Upgrade"
```
Comment (yogesh-mane) [inline suites/quincy/cephfs/tier-0_cephfs_upgrade_53_to_6x.yaml:139]: Could we change name to "Metadata collection before upgrade"

**Comment by yogesh-mane** [comment]
Also at  metadata_version_validation.py file's line 94 & 101 , output is none. Please check

**Comment by julpark-rh** [comment]
> Also at metadata_version_validation.py file's line 94 & 101 , output is none. Please check

Those two `None` are trying to print out result of the command. 

**None** and **not None** automatically detects if the cluster status is before or after upgrade.

out0, err0 = client.exec_command(
            sudo=True, cmd=f"stat {file_stat_path}", check_ec=False
        )
If the cmd returns None: before upgrade
if the cmd return not None: after upgrade

So returning those Nones are expected.

File: suites/quincy/cephfs/tier-0_cephfs_upgrade_53_to_6x.yaml (hunk 1 above)
Comment (julpark-rh) [reply suites/quincy/cephfs/tier-0_cephfs_upgrade_53_to_6x.yaml:179]: actually it is a good idea. somehow it caused your confusion.

Comment (AmarnatReddy) [reply suites/quincy/cephfs/tier-0_cephfs_upgrade_53_to_6x.yaml:179]: Updated the file Yogesh

**Review by hkadam134 — APPROVED** [review]
Looks Good to me.

Code review was done in the meeting

**Review by hkadam134 — APPROVED** [review]
LGTM

**Pull request merged**